modules/kargoBidAdapter.js

```diff
@@ -0,0 +1,166 @@
+import * as utils from 'src/utils';
+import {config} from 'src/config';
+import {registerBidder} from 'src/adapters/bidderFactory';
+const BIDDER_CODE = 'kargo';
+const HOST = 'https://krk.kargo.com';
+export const spec = {
+  code: BIDDER_CODE,
+  aliases: [],
+  isBidRequestValid: function(bid) {
+    if (!bid || !bid.params) {
+      return false;
+    }
+    return !!bid.params.placementId;
+  },
+  buildRequests: function(validBidRequests, bidderRequest) {
+    const currency = config.getConfig('currency');
+    const transformedParams = Object.assign({}, {
+      timeout: bidderRequest.timeout,
+      currency: currency,
+      cpmGranularity: 1,
+      cpmRange: {
+        floor: 0,
+        ceil: 20
+      },
+      adSlotIds: utils._map(validBidRequests, bid => bid.params.placementId)
+    }, spec._getAllMetadata());
+    const encodedParams = encodeURIComponent(JSON.stringify(transformedParams));
+    return Object.assign({}, bidderRequest, {
+      method: 'GET',
+      url: `${HOST}/api/v1/bid`,
+      data: `json=${encodedParams}`,
+      currency: currency
+    });
+  },
+  interpretResponse: function(adUnits, bidRequest) {
+    let bids = {};
+    utils._each(bidRequest.bids, bid => bids[bid.params.placementId] = bid);
+    const bidResponses = [];
+    for (let adUnitId in adUnits) {
+      let adUnit = adUnits[adUnitId];
+      bidResponses.push({
+        requestId: bids[adUnitId].bidId,
+        bidderCode: spec.code,
+        cpm: Number(adUnit.cpm),
+        width: adUnit.width,
+        height: adUnit.height,
+        ad: adUnit.adm,
+        ttl: 300,
+        creativeId: adUnitId,
+        netRevenue: true,
+        currency: bidRequest.currency
+      });
+
+      if (adUnit.receivedTracker) {
+        let el = document.createElement('img');
+        el.src = adUnit.receivedTracker;
+        document.body.appendChild(el);
```

Comment:
When do user syncs fire? We can just leave this off if it's not an immediate thing... we use this tracker to log that an ad unit was received by the page but if it's not fired in a deterministically "soon" fashion then it's better we just leave it off.

Reply:
Aha... ok. Not a user sync at all, then.

Yeah... there's no real good way to do that. Publishers get annoyed by libraries making extra requests because of the impact it has on page load times.

You could use the user sync module if you want... by default, we fire them 3 seconds after the auction has completed.

However, it still wouldn't be reliable. Since it affects the publishers' page performance, Prebid made this configurable... so they can limit the number that fire, opt out totally, etc.

Reply:
Ah okay cool, I will remove it then

Reply:
Sure thing.

If this is a feature you guys really need, I'd recommend opening an issue about it. I don't know how the Prebid.JS leadership feels about supporting this, but that's a good way to get discussion going with some more eyes on it.

Reply:
I've created a ticket here https://github.com/prebid/Prebid.js/issues/1762 but it's not urgent for us to have this support and I've just removed the code